client/src/components/menu-views/menu-view.js

```diff
@@ -58,24 +65,76 @@ class Menu extends Component {
         position="center"
         handlePageChange={this.handlePageChange}
       />
+
     );
   }
 
   initMenuItemDish() {
     const { menuitemdishes } = this.state;
     return (
-      <MenuItemDish menuitemdishes={menuitemdishes} />
+      <MenuItemDish
+        menuitemdishes={menuitemdishes}
+        addDishToOrderList={this.addDishToOrderList}
+        addMenuItemDishToOrderList={this.addMenuItemDishToOrderList}
+      />
     );
   }
 
+  addDishToOrderList(dishCategory, dishName) {
+    const previousDishArray = this.state.dishes;
+    const dishes = [...previousDishArray];
+    const dish = ` ${dishCategory} ${dishName}`;
+    dishes.push(dish);
+    this.setState({
+      dishes,
+    });
+  }
+
+  addMenuItemDishToOrderList(newMenuItemDish) {
+    const { menuItemDishesMap } = this.state;
+    let quantity = 1;
+    if (menuItemDishesMap.has(newMenuItemDish)) {
+      quantity = menuItemDishesMap.get(newMenuItemDish) + 1;
+    }
+    menuItemDishesMap.set(newMenuItemDish, quantity);
+    this.setState({
+      menuItemDishesMap,
+    });
+  }
+
   render() {
-    const { isFetching } = this.state;
+    const { id, name } = this.props;
+    const { isFetching, dishes, menuItemDishesMap } = this.state;
     if (isFetching) {
       return (
         <Container className="menu">
           <Header />
           {this.initMenuItemDish()}
           {this.initPagination()}
+          <Link to={{
+            pathname: `/restaurants/${id}/new-order`,
+            state: {
+              restaurantName: name,
+              dishes: dishes,
+              menuItemDishesMap: menuItemDishesMap,
+            },
+          }}
+          >
+            {this.state.dishes.length > 0 && (
+              <Button
+                className="complete"
+                variant="primary"
+                style={{
+                  marginRight: 40,
```

Comment:
One of possible decision
```
<div style={{
            display: 'flex',
            justifyContent: 'flex-end',
          }}
          >
            <Link to={{
              pathname: `/restaurants/${id}/new-order`,
              state: {
                restaurantName: name,
                dishes: dishes,
                menuItemDishesMap: menuItemDishesMap,
              },
            }}
            >
              {this.state.dishes.length > 0 && (
                <Button
                  className="complete"
                  variant="primary"
                >
                  Complete order (
                  {this.state.dishes.length}
                  {' '}
                  items)
                </Button>
              )}
            </Link>
          </div>
```

Reply:
Thanks, fixed!
Also fixed register user bug. But, the issue with refreshing app still exists.